packages/api/src/external/fhir/shared/hydrate-bundle.ts

```diff
@@ -0,0 +1,48 @@
+import { Extension, Organization, Patient } from "@medplum/fhirtypes";
+import { isValidUuid } from "@metriport/shared";
+import { metriportDataSourceExtension } from "@metriport/core/external/fhir/shared/extensions/metriport";
+import { uuidv7 } from "@metriport/core/util/uuid-v7";
+import { Bundle as ValidBundle } from "../../../routes/medical/schemas/fhir";
+
+/**
+ * Adds the Metriport and Document extensions to all the provided resources, ensures that all resources have UUIDs for IDs,
+ * and adds the Patient and Organization resources to the Bundle
+ */
+export function hydrateBundle(
+  bundle: ValidBundle,
+  patient: Patient,
+  org: Organization,
+  fhirBundleDestinationKey: string
+): ValidBundle {
+  const docExtension: Extension = {
+    url: "https://public.metriport.com/fhir/StructureDefinition/doc-id-extension.json",
+    valueString: fhirBundleDestinationKey,
+  };
+
+  const bundleWithExtensions = ensureUuidsAndAddExtensions(bundle, docExtension);
+  bundleWithExtensions.entry?.push({ resource: patient });
+  bundleWithExtensions.entry?.push({ resource: org });
+
+  return bundleWithExtensions;
+}
+
+type ReplacementIdPair = { old: string; new: string };
+
+function ensureUuidsAndAddExtensions(bundle: ValidBundle, docExtension: Extension): ValidBundle {
+  const replacements: ReplacementIdPair[] = [];
+  bundle.entry.forEach(entry => {
+    const oldId = entry.resource.id;
+    if (!isValidUuid(oldId)) {
+      replacements.push({
+        old: oldId,
+        new: uuidv7(),
+      });
+    }
+    entry.resource.extension = [metriportDataSourceExtension, docExtension];
```

Comment:
do we use this in the CDA we generate? 
also extension is an array so customer could theoretically provide something here, and we should overwrite if they have something there but instead just push to the array

Reply:
we don't use the extensions at all in CDAs.. but these will be useful to manipulate the FHIR resources on our end